cmd/ci-operator/main.go

```diff
@@ -625,11 +627,70 @@ func (o *options) resolveInputs(steps []api.Step) error {
 	// TODO: instead of mutating this here, we should pass the parts of graph execution that are resolved
 	// after the graph is created but before it is run down into the run step.
 	o.jobSpec.Namespace = o.namespace
-	log.Printf("Using namespace %s", o.namespace)
+
+	//None is a blocking error: the only impact is that the console URL is missing in the job log
+	var clusterConsoleURL string
+	cm, err := getConsoleConfigMap(o.clusterConfig)
+	if err != nil {
+		log.Printf("could not get ConfigMap console-config/openshift-console: %v", err)
+	} else {
+		clusterConsoleURL, err = getConsoleURL(cm)
+		if err != nil {
+			log.Printf("could not get console URL: %v", err)
+		}
+	}
+
+	if clusterConsoleURL != "" {
+		log.Printf("Using namespace %s/k8s/cluster/projects/%s", clusterConsoleURL, o.namespace)
+	} else {
+		log.Printf("Using namespace %s", o.namespace)
+	}
 
 	return nil
 }
 
+func getConsoleConfigMap(config *rest.Config) (*coreapi.ConfigMap, error) {
+	cl, err := client.New(config, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	cm := &coreapi.ConfigMap{}
+	err = cl.Get(context.Background(), client.ObjectKey{
+		Namespace: "openshift-console",
+		Name:      "console-config",
+	}, cm)
+	if err != nil {
+		return nil, err
+	}
+	return cm, nil
+}
+
+func getConsoleURL(cm *coreapi.ConfigMap) (string, error) {
+	if cm == nil {
+		return "", fmt.Errorf("nil config map")
+	}
+	consoleConfigString, ok := cm.Data["console-config.yaml"]
+	if !ok {
+		return "", fmt.Errorf("no key console-config.yaml in the data map")
+	}
+	consoleConfig := &ConsoleConfig{}
+	if err := yaml.Unmarshal([]byte(consoleConfigString), consoleConfig); err != nil {
```

Comment:
Re the route discussion, I think we can safely assume that the protocol will always be https, hence the route gives us everything we need

Reply:
Via `route.spec.host now`.